src/components/withNavigationFallback.tsx

```diff
@@ -0,0 +1,43 @@
+import React, {ComponentType, RefAttributes, ForwardedRef, ReactElement, forwardRef, useContext, useMemo} from 'react';
+import {NavigationContext} from '@react-navigation/core';
+import {NavigationProp} from '@react-navigation/native';
+import {ParamListBase} from '@react-navigation/routers';
+import getComponentDisplayName from '../libs/getComponentDisplayName';
+
+type AddListenerCallback = () => void;
+
+type RemoveListenerCallback = () => void;
+
+type NavigationContextValue = {
+    isFocused: () => boolean;
+    addListener: () => AddListenerCallback;
+    removeListener: () => RemoveListenerCallback;
+};
+
+export default function <TProps, TRef>(WrappedComponent: ComponentType<TProps & RefAttributes<TRef>>): (props: TProps & RefAttributes<TRef>) => ReactElement | null {
+    function WithNavigationFallback(props: TProps, ref: ForwardedRef<TRef>) {
+        const context = useContext(NavigationContext);
+
+        const navigationContextValue: NavigationContextValue = useMemo(() => ({isFocused: () => true, addListener: () => () => {}, removeListener: () => () => {}}), []);
```

Comment:
```suggestion
        const navigationContextValue: NavigationContextValue = useMemo(
            () => ({
                isFocused: () => true,
                addListener: () => () => {},
                removeListener: () => () => {},
            }),
            [],
        );
```

just to make it more readable :)